src/hotspot/share/runtime/lightweightSynchronizer.cpp

```diff
@@ -0,0 +1,1231 @@
+      }
+      task.done(current);
+      return true;
+    }
+    return false;
+  }
+
+  bool grow(JavaThread* current) {
+    ConcurrentTable::GrowTask grow_task(&_table);
+    if (run_task(current, grow_task, "Grow")) {
+      _table_size = table_size(current);
+      log_info(monitortable)("Grown to size: %zu", _table_size);
+      return true;
+    }
+    return false;
+  }
+
+  bool clean(JavaThread* current) {
+    ConcurrentTable::BulkDeleteTask clean_task(&_table);
+    auto is_dead = [&](ObjectMonitor** monitor) {
+      return (*monitor)->object_is_dead();
+    };
+    auto do_nothing = [&](ObjectMonitor** monitor) {};
+    NativeHeapTrimmer::SuspendMark sm("omworld");
+    return run_task(current, clean_task, "Clean", is_dead, do_nothing);
+  }
+
+  bool resize(JavaThread* current) {
+    LogTarget(Info, monitortable) lt;
+    bool success = false;
+
+    if (should_grow()) {
+      lt.print("Start growing with load factor %f", get_load_factor());
+      success = grow(current);
+    } else {
+      if (!_table.is_max_size_reached() && Atomic::load(&_resize)) {
+        lt.print("WARNING: Getting resize hints with load factor %f", get_load_factor());
+      }
+      lt.print("Start cleaning with load factor %f", get_load_factor());
+      success = clean(current);
+    }
+
+    Atomic::store(&_resize, false);
+
+    return success;
+  }
+
+  ObjectMonitor* monitor_put_get(Thread* current, ObjectMonitor* monitor, oop obj) {
+    // Enter the monitor into the concurrent hashtable.
+    ObjectMonitor* result = monitor;
+    Lookup lookup_f(obj);
+    auto found_f = [&](ObjectMonitor** found) {
+      assert((*found)->object_peek() == obj, "must be");
+      result = *found;
+    };
+    bool grow;
+    _table.insert_get(current, lookup_f, monitor, found_f, &grow);
+    verify_monitor_get_result(obj, result);
+    if (grow) {
+      try_notify_grow();
+    }
+    return result;
+  }
+
+  bool remove_monitor_entry(Thread* current, ObjectMonitor* monitor) {
+    LookupMonitor lookup_f(monitor);
+    return _table.remove(current, lookup_f);
+  }
+
+  bool contains_monitor(Thread* current, ObjectMonitor* monitor) {
+    LookupMonitor lookup_f(monitor);
+    bool result = false;
+    auto found_f = [&](ObjectMonitor** found) {
+      result = true;
+    };
+    _table.get(current, lookup_f, found_f);
+    return result;
+  }
+
+  void print_on(outputStream* st) {
+    auto printer = [&] (ObjectMonitor** entry) {
+       ObjectMonitor* om = *entry;
+       oop obj = om->object_peek();
+       st->print("monitor " PTR_FORMAT " ", p2i(om));
+       st->print("object " PTR_FORMAT, p2i(obj));
+       assert(obj->mark().hash() == om->hash(), "hash must match");
+       st->cr();
+       return true;
+    };
+    if (SafepointSynchronize::is_at_safepoint()) {
+      _table.do_safepoint_scan(printer);
+    } else {
+      _table.do_scan(Thread::current(), printer);
+    }
+  }
+};
+
+ObjectMonitorTable* LightweightSynchronizer::_omworld = nullptr;
```

Comment:
I preferred my version where ObjectMonitorTable was AllStatic which gets rid of _omworld-> everwhere, but the internal table is the pointer.

Reply:
I implemented the AllStatic table, and remove all mentions of OMWorld.